app/screens/channel_files/channel_files.tsx

```diff
@@ -0,0 +1,165 @@
+
+import {searchFiles} from '@actions/remote/search';
+import FileResults from '@components/files_search/file_results';
+import Loading from '@components/loading';
+import Search from '@components/search';
+import {General} from '@constants';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {FileFilter, FileFilters, filterFileExtensions} from '@utils/file';
+import {changeOpacity, getKeyboardAppearanceFromTheme} from '@utils/theme';
+
+import Header from './header';
+
+import type ChannelModel from '@typings/database/models/servers/channel';
+
+const TEST_ID = 'channel_files';
+
+type Props = {
+    channel: ChannelModel;
+    canDownloadFiles: boolean;
+    publicLinkEnabled: boolean;
+}
+
+const edges: Edge[] = ['bottom', 'left', 'right'];
+
+const styles = StyleSheet.create({
+    flex: {
+        flex: 1,
+    },
+    empty: {
+        alignItems: 'center',
+        flex: 1,
+        justifyContent: 'center',
+    },
+    list: {
+        paddingVertical: 8,
+    },
+    loading: {
+        justifyContent: 'center',
+        flex: 1,
+    },
+    searchBar: {
+        paddingHorizontal: 20,
+        paddingTop: 20,
+    },
+});
+
+const getSearchParams = (channelId: string, searchTerm?: string, filterValue?: FileFilter) => {
+    const term = `channel:${channelId}`;
+    const fileExtensions = filterFileExtensions(filterValue);
+    const extensionTerms = fileExtensions ? ' ' + fileExtensions : '';
+    const searchTerms = searchTerm ? ' ' + searchTerm : '';
+    return {
+        terms: term + searchTerms + extensionTerms,
+        is_or_search: true,
+    };
+};
+
+const emptyFileResults: FileInfo[] = [];
+
+function ChannelFiles({
+    channel,
+    canDownloadFiles,
+    publicLinkEnabled,
+}: Props) {
+    const theme = useTheme();
+    const serverUrl = useServerUrl();
+    const {formatMessage} = useIntl();
+    const searchTimeoutId = useRef<NodeJS.Timeout | null>(null);
+    const [filter, setFilter] = useState<FileFilter>(FileFilters.ALL);
+    const [loading, setLoading] = useState(true);
+    const [term, setTerm] = useState('');
+
+    const [fileInfos, setFileInfos] = useState<FileInfo[]>(emptyFileResults);
+
+    const handleSearch = useCallback(async (searchTerm: string, ftr: FileFilter) => {
+        const searchParams = getSearchParams(channel.id, searchTerm, ftr);
+        const {files} = await searchFiles(serverUrl, channel.teamId, searchParams);
+        setFileInfos(files?.length ? files : emptyFileResults);
+        setLoading(false);
+    }, [serverUrl, channel]);
+
+    useEffect(() => {
+        if (term) {
+            if (searchTimeoutId.current) {
+                clearTimeout(searchTimeoutId.current);
+            }
+
+            searchTimeoutId.current = setTimeout(() => {
+                handleSearch(term, filter);
+            }, General.SEARCH_TIMEOUT_MILLISECONDS);
+        } else {
+            handleSearch(term, filter);
+        }
+    }, [filter, term]);
```

Comment:
This may depend on `handleSearch`. 0/5 on whether we should trigger the effect if handleSearch changes (probably we do, since that would mean the channel or the server changed)